### PR TITLE
#1394 - Dynamic configuration of droppingItem

### PR DESCRIPTION
**Pull request by JamesRamm**
For issue #1394 
Idea here is to allow the `droppingItem` to be dynamically adjusted based on what is being dragged over.
Question - maybe we should prevent the `i` property from being overwritten (and limit it to `w` and `h`). I am not sure but dynamically changing `i` may cause issues for removing the placeholder. I don't see any use case for needing to change `i` anyway.

Example, in my particular use case the `onDropOver` handler may look like:

```js
const getDragData = e => e.dataTransfer.getData('dragData');
const evolveKeys => ({ minW: w, minH: h }) => ({ w, h })
const onDropOver = R.tryCatch(R.compose(evolveKeys, JSON.parse, getDragData),  () => ({ }));
```

### CODE REVIEW

**Comment by JamesRamm** [comment]
Hello maintainers? Any reply on this PR. It is not stale - still perfectly valid and compatible.

**Comment by STRML** [comment]
Thanks, happy to merge this if you could add a brief test & README update.

**Comment by STRML** [comment]
@JamesRamm something to think about - we usually allow returning `false` from handlers like this, too, so that the user can choose to disable the functionality based on some logic. I'll add a suggestion, let me know if you think it is useful.

File: lib/ReactGridLayout.jsx [hunk 1]
```diff
@@ -594,13 +594,16 @@ export default class ReactGridLayout extends React.Component<Props, State> {
...
       droppingItem,
+      onDragOver,
       margin,
       cols,
       rowHeight,
       maxRows,
       width,
       containerPadding
     } = this.props;
+    const finalDroppingItem = onDragOver ? { ...droppingItem, ...onDragOver(e) } : droppingItem;
```
Comment (STRML) [inline lib/ReactGridLayout.jsx:605]:
```suggestion
    // Allow user to customize the dropping item or short-circuit the drop based on the results
    // of the `onDragOver(e: Event)` callback.
    const onDragOverResult = onDragOver?.(e);
    if (onDragOverResult === false) return;
    const finalDroppingItem = { ...droppingItem, ...onDragOverResult };
```
```

**Comment by wuzhong1030** [comment]
Thank you!
It's commit very helpful for me.

**Comment by STRML** [comment]
Apologies, updated in pr/1395 and accidentally not pushed before merge. 415172f contains these changes.

**Comment by sayo96** [comment]
@JamesRamm  I'm unable to get the dragData through event.dataTransfer inside the onDropDragOver . Is there any way to implement this

**Comment by ekuzu** [comment]
@JamesRamm is it possible to get dataTransfer object inside onDropDragOver event?

**Comment by JamesRamm** [comment]
@sayo96  @ekuzu 
The event is `onDragOver`, not `onDropDragOver`

**Comment by sayo96** [comment]
@JamesRamm I need the event `onDropDragOver`. If possible please look at this issue i ran into in firefox https://github.com/react-grid-layout/react-grid-layout/issues/1841

**Comment by sonmansu** [comment]
> @sayo96 @ekuzu The event is `onDragOver`, not `onDropDragOver`

`GridLayout` does not have `onDragOver` props.
How can I get `dataTransfer` object inside `onDropDragOver`or `onDragOver`?

**Comment by B33fb0n3** [comment]
If you add data inside the droppable-element, that data is not accessable inside the onDragOver (onDropDragOver prop).

Add Data:
```jsx
onDragStart={e => {
 e.dataTransfer.setData("text/plain", JSON.stringify(component.blockData));
}}
```

But if you try to access this data via 

```jsx
const eventData = e.dataTransfer.getData("text/plain");
```
It returns null ^^

If you access it the same way while handling the drop event, you get the data

**Comment by szabikr** [comment]
Any updates on this?

I would also like to get the `dataTransfer` data from the event in the `onDropDragOver` handler but get an empty string.

Thank you!

**Comment by mm105** [comment]
Am I missing something or there is still no option to have size of dragged item while dragging over grid to dynamically determine size of palceholder?

**Pull request merged**